maps.Rmd

```diff
@@ -299,52 +299,140 @@ ggplot(dawson[-69]) +
 
 <!-- HW: do you have any thoughts on a modern approach to this problem? -->
 
-Instead of displaying context with vector boundaries, you might want to draw a traditional map underneath. This is called a raster image. The easiest way to get a raster map of a given area is to use the ggmap package, which allows you to get data from a variety of online mapping sources including OpenStreetMap and Google Maps. Downloading the raster data is often time consuming so it's a good idea to cache it in a rds file, as illustrated here: \index{ggmap} \index{Raster data}
+A second way to supply geospatial information for mapping is to rely on **raster data**. Unlike the simple features format, in which geographical entities are specified in terms of a set of lines, points and polygons, rasters take the form of images. In the simplest case raster data might be nothing more than a bitmap file, but there are many different image formats out there. In the geospatial context specifically, there are image formats that include metadata (e.g., geodetic datum, coordinate reference system) that can be used to map the image information to the surface of the Earth. For example, one common format is GeoTIFF, which is a regular TIFF file with additional metadata supplied. Happily, most formats can be easily read into R with the assistance of GDAL (the Geospatial Data Abstraction Library, https://gdal.org/). For example the sf package contains a function `sf::gdal_read()` that provides access to the GDAL raster drivers from R. However, you rarely need to call this function directly, as there are other high level functions that take care of this for you. 
+
+As an illustration, suppose we wish to plot satellite images made publicly available by the Australian Bureau of Meterorology (BOM) on their FTP server. The bomrang package [@bomrang] provides a convenient interface to the server, including a `get_available_imagery()` function that returns a vector of filenames and a `get_satellite_imagery()` function that downloads a file and imports it directly into R. For expository purposes, however, I'll use a more flexible method that could be adapted to any FTP server, and use the `curl_download()` function from the curl package [@curl]
```

Comment:
Any reason to prefer `curl_download()` over `download.file()`?

Reply:
I vaguely recall it was because I was imitating `bomrang::get_satellite_imagery()` which uses `curl_download()`. I'm guessing the perceived advantage is that `curl_download()` writes straight to disk which seems like a good idea for large files. I guess it doesn't matter in this case because the files are <4Mb in size, but on the other hand if someone does cut and paste from the code in this example and the files they're downloading are large, I suppose I'd rather they use `curl_download()`. Possibly I'm overthinking it. I'm happy either way really. It's an easy fix though - let me know which one you prefer!

Reply:
`download.file()` writes straight to disk here; I'd have a slight preference for it since it's a base function so I think slightly more people might be familiar with it.